src/target/llvm/codegen_llvm.cc

```diff
@@ -736,7 +736,40 @@ llvm::Function* CodeGenLLVM::GetIntrinsicDecl(llvm::Intrinsic::ID id, llvm::Type
 #endif  // TVM_LLVM_VERSION
 }
 
+// Check if this is a warp shuffle intrinsic call and match its
+// corresponding nvvm intrinsic. Return true if the match is successful.
+static bool GetWarpShuffleIntrinsic(const CallNode* op, llvm::Intrinsic::ID* id) {
+  // Only 32 bit data type is supported.
+  if (op->dtype.is_vector() || op->dtype.bits() != 32) {
```

Comment:
We should move this to the codegen_nvptx.cc instead and overload the intrinsics creation.

Reply:
Sure. I will have a look.

Reply:
I'll can do this move when I fix rocm. (I needed to do it anyways, so...)